.github/workflows/test-windows.yml

```
@@ -64,20 +88,89 @@ jobs:
           
   profiler-tests:
     runs-on: windows-2022
-    # Disable profiler tests for now
-    if: ${{ false }} 
+    needs: [ 'format', 'tests' ]
     steps:
       - uses: actions/checkout@v4
       - name: Bootstrap Action Workspace
         uses: ./.github/workflows/bootstrap
         with:
           rust: 'true'
+          tc-cloud: ${{ secrets.TC_CLOUD_TOKEN }}
+
+      - name: 'Tests: Profiler'
+        run: ./build.bat test --test-suite profiler
+        
+  test-iis:
+    runs-on: windows-latest
+    needs: [ 'format', 'tests' ]
+    
+    steps:
+      - uses: actions/checkout@v4
+      - name: Bootstrap Action Workspace
+        uses: ./.github/workflows/bootstrap
 
-      - name: Setup Testcontainers Cloud Client
-        if: github.event_name != 'pull_request' || github.event_name == 'pull_request' && github.event.pull_request.head.repo.fork == false
-        uses: atomicjar/testcontainers-cloud-setup-action@v1
+      - uses: actions/cache@v4
         with:
-          token: ${{ secrets.TC_CLOUD_TOKEN }}
+          path: ~/.nuget/packages
+          key: "${{ runner.os }}-nuget-${{ hashFiles('**/*.[cf]sproj*') }}"
 
-      - name: 'Tests: Profiler'
-        run: ./build.bat test --test-suite profiler
\ No newline at end of file
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v2
```

pin version when using third-party actions. dependabot will bump them.

```suggestion
        uses: microsoft/setup-msbuild@6fb02220983dee41ce7ae257b6f4d8f9bf5ed4ce # v2
```